src/Spectre.libGenetic.Tests/CrossoverOperatorTest.cpp

```diff
@@ -0,0 +1,158 @@
+/*
+* CrossoverOperatorTest.cpp
+* Tests crossover operator.
+*
+Copyright 2017 Grzegorz Mrukwa
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#define GTEST_LANG_CXX11 1
+
+#include <gtest/gtest.h>
+#include <numeric>
+#include "Spectre.libGenetic/CrossoverOperator.h"
+#include "Spectre.libGenetic/InconsistentChromosomeLengthException.h"
+
+namespace
+{
+using namespace Spectre::libGenetic;
+
+TEST(CrossoverOperatorInitialization, initializes)
+{
+    CrossoverOperator crossover(0);
+}
+
+class CrossoverOperatorTest: public ::testing::Test
+{
+public:
+    CrossoverOperatorTest(){}
+protected:
+    const unsigned NUMBER_OF_TRIALS = 1000;
+    const double ALLOWED_MISS_RATE = 0.05;
+    const Seed SEED = 0;
+    const Individual true_individual = Individual({ true, true, true, true });
+    const Individual false_individual = Individual({ false, false, false, false });
+    CrossoverOperator crossover;
+
+    void SetUp() override
+    {
+        crossover = CrossoverOperator(SEED);
+    }
+};
+
+TEST_F(CrossoverOperatorTest, throws_on_inconsistent_sizes)
+{
+    const Individual shorter({ false });
+    const Individual longer({ false, true });
+    EXPECT_THROW(crossover(shorter, longer), InconsistentChromosomeLengthException);
+}
+
+TEST_F(CrossoverOperatorTest, child_has_the_same_size)
+{
+    const auto parentsSize = true_individual.size();
+    const auto child = crossover(true_individual, false_individual);
+    EXPECT_EQ(child.size(), parentsSize);
+}
+
+TEST_F(CrossoverOperatorTest, crossover_of_same_parents_result_in_copy)
+{
+    const auto singleParent = true_individual;
+    for(unsigned i = 0; i < NUMBER_OF_TRIALS; ++i)
+    {
+        const auto child = crossover(singleParent, singleParent);
+        for (size_t j = 0; j < child.size(); ++j)
+        {
+            EXPECT_EQ(child[j], singleParent[j]);
+        }
+    }
+}
+
+TEST_F(CrossoverOperatorTest, cuts_with_symmetric_distribution)
+{
+    const auto expectedCounts = NUMBER_OF_TRIALS * true_individual.size() * 0.5;
```

Comment:
I'm just expressing mathematically number of the expected counts and using `auto` to not to care about its type at all. It doesn't hurt in this case, as GTest is doing the cast under the hood.